lnwallet/channel.go

```diff
@@ -291,6 +320,100 @@ type PaymentDescriptor struct {
 	isForwarded bool
 }
 
+// PayDescsFromRemoteLogUpdates converts a slice of LogUpdates received by a
+// the remote peer into PaymentDescriptors to inform a link's forwarding
+// decisions.
+func (lc *LightningChannel) PayDescsFromRemoteLogUpdates(height uint64,
+	logUpdates []channeldb.LogUpdate) []*PaymentDescriptor {
+
+	lc.Lock()
+	defer lc.Unlock()
+
+	var payDescs []*PaymentDescriptor
+	for i, logUpdate := range logUpdates {
+		var pd *PaymentDescriptor
+		switch wireMsg := logUpdate.UpdateMsg.(type) {
+
+		// For offered HTLC's, we'll map that to a PaymentDescriptor
+		// with the type Add, ensuring we restore the necessary fields.
+		// From the PoV of the commitment chain, this HTLC was included
+		// in the remote chain, but not the local chain.
+		case *lnwire.UpdateAddHTLC:
+			// First, we'll map all the relevant fields in the
+			// UpdateAddHTLC message to their corresponding fields in the
+			// PaymentDescriptor struct. We also set addCommitHeightRemote
```

Comment:
Looks like the comment was copied from elsewhere in the file. It's a bit off though as the `addCommitHeightRemote` field isn't set anywhere.

Reply:
Fixed.